### DISCUSSION TITLE
使用ssp、wcc等算法时，边必须带权重吗？

### DISCUSSION THREAD

**Discussion by xglreal** [alibaba/GraphScope#2584] [Q&A] [answered]
import graphscope
from graphscope.dataset import load_p2p_network
sess = graphscope.session(cluster_type="hosts", mode="eager")
g = load_p2p_network(sess)
pg = g.project(vertices={"host": ["id"]}, edges={"connect": ["dist"]})
c = graphscope.hits(pg, tolerance=0.01, max_round=10, normalized=True)
sess.close()

必须要有dist属性吗？

**Comment by siyuan0322** [comment, marked as answer]
依具体算法需不需要用到边的属性而定。 WCC 不需要。 SSSP 需要。具体可以根据算法的源码看到有没有用到属性。
